src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -110,6 +110,8 @@ private ISymbolNode CreateReadyToRunHelperCell(ReadyToRunHelper helperId)
             SignatureContext signatureContext, 
             bool isUnboxingStub = false)
         {
+            ThrowIfTypeRequiresRuntimeJit(targetMethod.OwningType);
```

Comment:
Does this check kick even when the codegen does not actually need the type? E.g. `sizeof(Vector<int>)` or method that takes `sizeof(Vector<int>)` argument that is not otherwise used?

Reply:
As I mentioned in the commit description, right now I don't know what code path needs fixing to cater for proper sizeof management. I speculate this is probably one of the CorInfoImpl code paths. I guess it would be too big a hammer to fail all calls to resolveToken resolving to the Vector types.

Reply:
> I guess it would be too big a hammer to fail all calls to resolveToken resolving to the Vector types.

It is actually what crossgen does today.

Reply:
OK, thanks for the clarification, I have added similar provisions to my pending change.